components/serializers.py

```diff
@@ -72,8 +74,38 @@ class Meta:
             "catalog_data",
             "component_data",
             "project_data",
+            "component_json",
         )
 
+    def partial_update(self, instance, validated_data):
+        if controls := validated_data.get("controls"):  # && controls not in:
+            print(controls)
+            # instance.controls.append(controls)
+            instance.controls = list(set(instance.controls).union(controls))
+
+        if description := validated_data.get("description"):
+            implemented_requirement = {
+                "uuid": str(uuid.uuid4()),
+                "props": [
+                    {
+                        "name": "security_control_type",
+                        "uuid": str(uuid.uuid4()),
+                        "value": "Hybrid",
+                    },
+                    {"name": "provider", "uuid": str(uuid.uuid4()), "value": "No"},
```

Comment:
Each time you call `uuid.uuid4` you'll get a different id. Was it intentional that each of these fields had a separate uuid?